cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/RemoteClient.java

```diff
@@ -139,6 +146,41 @@ public HttpResponse execute(HttpRequest request) throws IOException, Unauthorize
     }
   }
 
+  /**
+   * Makes a streaming {@link HttpRequest} and consumes the response using the {@link HttpContentConsumer} provided
+   * in the request. It retries on failure.
+   */
+  public void executeStreamingRequest(HttpRequest request) throws IOException, UnauthorizedException {
+    URL rewrittenURL = rewriteURL(request.getURL());
+    Multimap<String, String> headers = request.getHeaders();
+    headers = headers == null ? HashMultimap.create() : HashMultimap.create(headers);
+
+    // Add Authorization header and use a rewritten URL if needed
+    RemoteAuthenticator authenticator = getAuthenticator();
+    if (authenticator != null && headers.keySet().stream().noneMatch(HttpHeaders.AUTHORIZATION::equalsIgnoreCase)) {
+      setAuthHeader(headers::put, HttpHeaders.AUTHORIZATION, authenticator.getType(), authenticator.getCredentials());
+    }
+
+    internalAuthenticator.applyInternalAuthenticationHeaders(headers::put);
+
+    HttpRequest httpRequest =
+      new HttpRequest(request.getMethod(), rewrittenURL, headers, request.getBody(), request.getBodyLength(),
+                      request.getConsumer());
+    Retries.callWithRetries(() -> {
+      HttpResponse httpResponse = HttpRequests.execute(httpRequest, httpRequestConfig);
+
+      if (httpResponse.getResponseCode() != HttpURLConnection.HTTP_OK) {
+        throw new IOException(httpResponse.getResponseBodyAsString());
+      }
+      try {
+        httpResponse.consumeContent();
+      } catch (Exception e) {
+        LOG.warn("No content from http response ", e);
+      }
+      return null;
+    }, RetryStrategies.limit(RETRY_LIMIT, RetryStrategies.fixDelay(RETRY_DELAY, TimeUnit.SECONDS)));
```

Comment:
Overall I don't think other RemoteClient calls do retries. It's betters not to do it to be consistent. If caller needs it, it can do it.